### PR TITLE
Nested Fields for has-many association

### LINKED ISSUE #38: Nested Fields
It would be great to support nested fields. 

It appears Administrate has a plugin for this as well: https://github.com/nickcharlton/administrate-field-nested_has_many

### PR DESCRIPTION
**Pull request by dhurba87**
fixes #38 

## What type of PR is this?
- [x] Feature
- [x] Unit testing
- [ ] Documentation Update

## Description
Allows creating nested form fields

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -27,3 +27,4 @@ gem "sass-rails", ">= 6"
 gem "pg"
 gem "mysql2"
 gem "sqlite3"
+gem "cocoon"
```
Comment (dhurba87) [inline Gemfile]: @excid3 I have used cocoon gem for dynamic nested form. I can see that we're not using jQuery so is it okay if I use [cocoon-vanilla-js](https://github.com/oddcamp/cocoon-vanilla-js), which is an equivalent JS written in vanilla-js?

Comment (excid3) [reply Gemfile]: I think we can do it without cocoon. I did a screencast on that a bit ago: https://gorails.com/episodes/dynamic-nested-forms-with-stimulus-js

Comment (dhurba87) [reply Gemfile]: Looks like screencast is not available for me but I'll try to implement it using stimulusjs. Thanks :)

Comment (excid3) [reply Gemfile]: The source code is linked in the notes on that page: https://github.com/gorails-screencasts/dynamic-nested-forms-with-stimulusjs

**Pull request closed without merging**